FhirToDataLake/SynapseValidate/Synapse_Validator.py

```diff
@@ -0,0 +1,207 @@
+    Headers = {'Accept': 'application/fhir+json', "Prefer": "respond-async"}
+    Start_Datetime = "1970-01-01T00:00:00-00:00"
+
+    def __init__(self, baseUrl):
+        self.baseUrl = baseUrl
+
+    def get_all_entries(self, resource_type):
+        end_datetime = datetime.today().strftime('%Y-%m-%dT%H:%M:%S-00:00')
+        url = self.baseUrl + f"/{resource_type}?_lastUpdated=ge{RawDataSource.Start_Datetime}&_lastUpdated=lt{end_datetime}&_sort=_lastUpdated&_count=1000"
+        historical_data = self.get_entries(url)
+
+        '''
+        url = self.baseUrl + f"/{resource_type}?_lastUpdated=ge{end_datetime}&_lastUpdated=lt{end_datetime}&_sort=_lastUpdated&_count=1000"
+        latest_data = self.get_entries(url)
+        return historical_data + latest_data
+        '''
+        return historical_data
+
+
+    def get_entries(self, url):
+        response = requests.get(url, headers=RawDataSource.Headers)
+        if response.status_code != requests.codes.ok:
+            raise Exception(f"Get data from {url} failed: " + response.text)
+
+        response_content = response.content
+        response_object = json.loads(response_content.decode('utf-8'))
+        if "entry" not in response_object:
+            return []
+
+        sumData = []
+        sumData.extend(response_object["entry"])
+        continuation_token = self.parse_continuation_token(response_object)
+
+        while continuation_token:
+            response = requests.get(url, headers=RawDataSource.Headers, params={"ct": continuation_token})
+            if response.status_code != requests.codes.ok:
+                raise Exception(f"Get data from {url} failed: " + response.text)
+
+            response_content = response.content
+            response_object = json.loads(response_content.decode('utf-8'))
+            if "entry" not in response_object:
+                continuation_token = None
+            else:
+                sumData.extend(response_object["entry"])
+                continuation_token = self.parse_continuation_token(response_object)
+        return sumData
+
+    @staticmethod
+    def parse_continuation_token(response_content):
+        for link_item in response_content["link"]:
+            if link_item["relation"] == "next":
+                parsed_url = urlparse.urlparse(link_item["url"])
+                return urlparse.parse_qs(parsed_url.query)["ct"][0]
+        return None
+
+
+class SchemaManager:
+    def __init__(self, schema_collection_directory):
+        self.schemas = SchemaManager.load_schemas(schema_collection_directory)
+
+    @staticmethod
+    def load_schemas(schema_collection_directory) -> dict:
+        schemas = {}
+        for schema_file in os.listdir(schema_collection_directory):
+            with open(os.path.join(schema_collection_directory, schema_file)) as f:
+                schema = json.load(f)
+                schemas[schema["Type"]] = schema
+        return schemas
+
+    def generate_field_mapping_dicts(self):
```

Comment:
What does `field_mapping_dicts` mean?

Reply:
A field dict from Json schema to SQL schema.
E.g.
Json path <Patient.name.family> becomes column "Patient.name.family"

Reply:
Maybe we can consider a better variable name for it?

Reply:
How about ```column_dicts```, this is designed for future detailed query check?

Reply:
Updated to ```column_dict```